packages/ui/src/components/CognitoAuth/components/ForgotPasswordView/index.tsx

```diff
@@ -0,0 +1,130 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+import { FC, useCallback, useMemo, useState } from 'react';
+import FormRenderer, { Schema, componentTypes } from '../../../FormRenderer';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Button from '@cloudscape-design/components/button';
+
+export interface ForgotPasswordViewProps {
+    data?: any;
+    onResetPassword: (verificationCode: string, newPassword: string) => Promise<unknown>;
+    onBackToSignIn: () => void;
+}
+
+const ForgotPasswordView: FC<ForgotPasswordViewProps> = ({ onResetPassword, onBackToSignIn, data }) => {
+    const [errorMessage, setErrorMessage] = useState();
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const schema = useMemo(() => {
```

Comment:
Same here, as i can see, this calculation is not relay on `data` and this just a simple object declaration that doesn't involve any expensive computation

Reply:
The schema only depends on the data. But if it is not enclosed in useMemo, the schema object itself will change. All the FormRenderer will re-render